From cdb382de380c5edab8a18bc5eb4da5110ee015db Mon Sep 17 00:00:00 2001
From: Ralf Ramsauer <ralf.ramsauer@oth-regensburg.de>
Date: Wed, 7 Jun 2017 15:48:43 +0200
Subject: [PATCH 10/15] arm, arm64: export __hyp_stub_vectors

HVC_GET_VECTORS got removed. External hypervisors, like Jailhouse, need
this address when they are deactivated, in order to restore original
state.

Upstream-Status: Pending

Signed-off-by: Ralf Ramsauer <ralf.ramsauer@oth-regensburg.de>
Signed-off-by: Jan Kiszka <jan.kiszka@siemens.com>
(cherry picked from commit 04ea1cb888eece6aa4b7a0705e65f63be1f25b1a)
Signed-off-by: Ricardo Salveti <ricardo@foundries.io>
---
 arch/arm/kernel/hyp-stub.S   | 2 ++
 arch/arm64/kernel/hyp-stub.S | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/arch/arm/kernel/hyp-stub.S b/arch/arm/kernel/hyp-stub.S
index b19ae4459f14b4..b9ba0585a4ce27 100644
--- a/arch/arm/kernel/hyp-stub.S
+++ b/arch/arm/kernel/hyp-stub.S
@@ -6,6 +6,7 @@
 #include <linux/init.h>
 #include <linux/irqchip/arm-gic-v3.h>
 #include <linux/linkage.h>
+#include <asm-generic/export.h>
 #include <asm/assembler.h>
 #include <asm/virt.h>
 
@@ -259,4 +260,5 @@ __hyp_stub_trap:	W(b)	__hyp_stub_do_trap
 __hyp_stub_irq:		W(b)	.
 __hyp_stub_fiq:		W(b)	.
 ENDPROC(__hyp_stub_vectors)
+EXPORT_SYMBOL_GPL(__hyp_stub_vectors)
 
diff --git a/arch/arm64/kernel/hyp-stub.S b/arch/arm64/kernel/hyp-stub.S
index 2ee18c860f2ab6..47d117fd1fbca8 100644
--- a/arch/arm64/kernel/hyp-stub.S
+++ b/arch/arm64/kernel/hyp-stub.S
@@ -9,6 +9,7 @@
 #include <linux/init.h>
 #include <linux/linkage.h>
 
+#include <asm-generic/export.h>
 #include <asm/assembler.h>
 #include <asm/el2_setup.h>
 #include <asm/kvm_arm.h>
@@ -66,6 +67,7 @@ SYM_CODE_START(__hyp_stub_vectors)
 	ventry	el1_fiq_invalid			// FIQ 32-bit EL1
 	ventry	el1_error_invalid		// Error 32-bit EL1
 SYM_CODE_END(__hyp_stub_vectors)
+EXPORT_SYMBOL_GPL(__hyp_stub_vectors)
 
 	.align 11
 
-- 
2.40.1

